Desktop_Interface/isobuffer.cpp

```diff
@@ -2,391 +2,364 @@
+		voltageLevel += virtualParent->currentVmean;
+	}
 #ifdef INVERT_MM
-    if(virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
+	if (virtualParent->driver->deviceMode == 7) voltageLevel *= -1;
 #endif
-    if (virtualParent->driver->deviceMode != 7) voltageLevel -= voltage_ref;
+	if (virtualParent->driver->deviceMode != 7) voltageLevel -= voltage_ref;
 
-    //voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
-    sample = (voltageLevel * (frontendGain*scope_gain*TOP))/(vcc/2);
-    return sample;
+	//voltageLevel = (sample * (vcc/2)) / (frontendGain*scope_gain*TOP);
+	sample = (voltageLevel * (frontendGain*scope_gain*TOP))/(vcc/2);
+	return sample;
 }
 
 #define NUM_SAMPLES_SEEKING_CAP (20)
 
 #ifdef INVERT_MM
-    #define X0_COMPARISON_CAP >
-    #define X1_X2_COMPARISON_CAP <
+constexpr auto X0_COMP_FTOR = std::greater<int> {};
+constexpr auto X1_X2_COMP_FTOR = std::less<int> {};
 #else
-    #define X0_COMPARISON_CAP <
-    #define X1_X2_COMPARISON_CAP >
+constexpr auto X0_COMP_FTOR = std::less<int> {};
+constexpr auto X1_X2_COMP_FTOR = std::greater<int> {};
 #endif
 
-//For capacitance measurement.  x0, x1 and x2 are all various time points used to find the RC coefficient.
-int isoBuffer::cap_x0fromLast(double seconds, double vbot){
-    int samplesInPast = seconds * samplesPerSecond;
-    if(back < samplesInPast){
-        return -1; //too hard, not really important
-    }
-    short vbot_s = inverseSampleConvert(vbot, 2048, 0);
-    qDebug() << "vbot_s (x0) = " << vbot_s;
-
-    int num_found = 0;
-    for(int i=samplesInPast; i; i--){
-        short currentSample = buffer[back - i];
-        if(currentSample X0_COMPARISON_CAP vbot_s){
-            num_found++;
-        } else num_found--;
-        if(num_found < 0){
-            num_found = 0;
-        }
-        if (num_found > NUM_SAMPLES_SEEKING_CAP){
-            return samplesInPast-i;
-        }
-    }
-    return -1;
-}
+int capSample(isoBuffer const& self, int rbegin, int target, double seconds, double value, auto comp)
+{
+	int samples = seconds * self.samplesPerSecond;
 
-int isoBuffer::cap_x1fromLast(double seconds, int x0, double vbot){
-    int samplesInPast = seconds * samplesPerSecond;
-    samplesInPast -= x0;
-    if(back < samplesInPast){
-        return -1; //too hard, not really important
-    }
-    short vbot_s = inverseSampleConvert(vbot, 2048, 0);
-    qDebug() << "vbot_s (x1) = " << vbot_s;
-
-    int num_found = 0;
-    for(int i=samplesInPast; i; i--){
-        short currentSample = buffer[back - i];
-        if(currentSample X1_X2_COMPARISON_CAP vbot_s){
-            num_found++;
-        } else num_found--;
-        if(num_found < 0){
-            num_found = 0;
-        }
-        if (num_found > NUM_SAMPLES_SEEKING_CAP){
-            return samplesInPast-i + x0;
-        }
-
-    }
-    return -1;
-}
+	if (self.back < samples + rbegin) return -1;
+
+	short sample = self.inverseSampleConvert(value, 2048, 0);
```

Comment:
This "2048" value should probably be a defined constant in an anonymous namespace at the top of the .cpp.
It represents the top value of the multimeter ADC, so a name like "kTopMultimeter" would be good.  Also, you should be able to constexpr it.

Reply:
Ah, got it.